pytube/cipher.py

```diff
@@ -224,7 +224,8 @@ def get_transform_object(js: str, var: str) -> List[str]:
     regex = re.compile(pattern, flags=re.DOTALL)
     transform_match = regex.search(js)
     if not transform_match:
-        raise RegexMatchError(caller="get_transform_object", pattern=pattern)
+        logger.error(f"no match found for pattern: {pattern}")
+        return []
```

Comment:
I am skeptical about this part. 
Returning an empty list is not what this method is expected to do. It will crash as soon as `self.transform_plan[0]` is called in the Cipher constructor